### PR TITLE
Separate ProgressBar from RecursiveOptionParser

### PR DESCRIPTION
**Pull request by claudep**
This small refactoring is a step in making the RecursiveOptionParser class a bit lighter and more manageable.

### CODE REVIEW

File: translate/misc/optrecurse.py [hunk 1]
```diff
@@ -32,6 +32,36 @@
...
+        if progress_type in ('bar', 'verbose'):
+            self._progressbar = self.progress_types[progress_type](0, len(allfiles))
+            logger = logging.getLogger(os.path.basename(sys.argv[0])).getChild("progress")
+            logger.setLevel(logging.INFO)
+            logger.propagate = False
+            handler = logging.StreamHandler()
+            handler.setLevel(logging.INFO)
+            handler.setFormatter(logging.Formatter())
+            logger.addHandler(handler)
+            logger.info("processing %d files...", len(allfiles))
```
Comment (unho) [inline translate/misc/optrecurse.py]: Nit: calculate `len(allfiles)` once.

**Comment by unho** [comment]
lgtm

**Comment by unho** [comment]
@claudep I would have named it `file_count`, but thanks a lot for the contribution!

**Comment by claudep** [comment]
Sure, it's a better name: da129a4
It might be details, but good naming makes reading the code easier!

**Pull request merged**